int64 circlesNo
int64 trianglesNo
int64 linesNo
int64 squaresNo
